### ISSUE TITLE
[BUG] Poor accessibility for screenreaders

### ISSUE THREAD

**Issue by cyrmax** [marticliment/WingetUI#1767]
### Please confirm these before moving forward

- [X] I have searched for my issue and not found a work-in-progress/duplicate/resolved issue.
- [X] I have tested that this issue has not been fixed in the latest [(beta or stable) release](https://github.com/marticliment/WingetUI/releases/).
- [X] I have checked the [FAQ](https://github.com/marticliment/WingetUI#frequently-asked-questions) section for solutions
- [X] This issue is about a bug (if it is not, please use the [correct template](https://github.com/marticliment/WingetUI/issues/new/choose))

### Describe your issue

I'm completely blind and therefore use NVDA Screenreader.
This is a program which reads the contents of a computer screen with synthesized voice and allows non-visual interaction only with keyboard.

I have installed WingetUI 2.2.0 (latest version at the moment) and realized that non-visual accessibility of a program is poor.

* Some buttons are not reachable without mouse;
* Switches that should be a radio buttons are announced as checkboxes;
* Some buttons cannot be activated with `enter` and should be pressed either with `space` or only with mouse click;
* In treeviews with package lists NVDA reads only "Treeview item checked/unchecked" and does not read other information;
* Main app tabs (such as updates, discover, etc) should be real tabs but NVDA announces them as checkboxes;
* When the app does something heavy, such as updating many packages at once the screenreader is flooded with useless information and sometimes becomes unresponsive and crashes;
* Many buttons, especially in first setup windows, have no text labels at all and are announced just like "button" or "button return top".

I will try to investigate those problems and write my thoughts here but I really ask you to fix the accessibility. I think thousands of blind people around the world will thank you a lot.

### Steps to reproduce the issue

1. Install NVDA screenreader from nvaccess.org;
2. Install latest WingetUI (2.2.0 at the moment);
3. Start NVDA screenreader, disconnect your mouse and close your eyes;
4. Wear headphones if necessary;
5. Go through NVDA first setup and set speech parameters such as voice, rate and pitch;
6. Open WingetUI and try to use it only with keyboard and without eyes.

### WingetUI Log

```text
Not applicable.
```

### Package Managers Logs

```text
Not applicable.
```

**Comment by marticliment** [comment]
I am aware of this issue, and the ongoing WingetUI redesign plans to fix this issue. For more info see #1724

**Comment by cyrmax** [comment]
Okay, I will try it out and tell if it is better now.

**Comment by cyrmax** [comment]
I have tested the main screen accessibility and recorded a screencast where you can hear what is happening and also probably read it on the screen. Also all focused elements are highlighted by my screenreader as I know.

The main problem is that some elements have their text not directly as an accessibility label but as a text label inside the element. For example these checkboxes which are read as "checkbox not checked" without any descriptive name label.
The same is for buttons.
And also I didn't understand what are these "grouping expanded" elements. Probably they are list views which are not read correctly.

https://drive.google.com/file/d/1-TQlgPbEcRZraGVmCb0f_omA2Q697Zxk/view?usp=sharing

Sorry, but for Github this file is too big (48 mb) and so I have uploaded it to my GDrive.

**Comment by marticliment** [comment]
Ok, I will fix this

**Comment by cyrmax** [comment]
Also the updates screen is illustrative.
Here we have a treeview and a list which should have their own accessibility labels, for example as I understand their purpose "Sources" and "Updatable packages".
And also we hear some element IDs instead of package names and data. We have to use advanced NVDA features to reach to the information and see if the package is checked or not.
What you could do is to merge all the information about the package into its accessibility label and provide a role of checkbox which will represent the current checkbox state.
Here is another screencast about this.

https://drive.google.com/file/d/1-UArDs8CF7aDYUYTnqEZJn-Fx5qrgGXK/view?usp=sharing

**Comment by marticliment** [comment]
Hello @cyrmax,

I have publised WingetUI 3.0.1 with a few little accessibility improvements, but my plan is to fully adapt WingetUI to be way more accesible in 3.1.0

**Comment by christopherpross** [comment]
@marticliment 
Thank you for your awesome work.
I am also a screenreader user and a c#-developer. If you have questions or need help to improve the accessibility, feel free to ask.

**Comment by britechguy** [comment]
Overall, UniGetUI is written with accessibility in mind, but it's now "the little things" that are show stoppers for independent use with a screen reader.  I am fully sighted, but am a tutor for those new to using screen readers.  I just did a "quick and dirty" run through of the UniGetUI main interface and can report that there are several unlabeled buttons as well as other controls.  I have not done a comprehensive navigation of the entire UI.

Here are the things I've found using NVDA over the last 10 minutes or so.  I suspect all would be easy to fix with changes in control labeling.

1. The filter toggle button immediately to the left of the Search for packages search edit box is announced as, "Toggle button," along with it's pressed/unpressed state.  But it needs to have its function announced, too, e.g., "Filter Toggle Button."
2. The actual search button associated with the search edit box is only announced as "button" when you land on it.  Again, the actual function of the control needs to accompany the announcement of the control type, in this case, "Search button."
3. The Reload button to the right of the search button is also only announced as "button."
4. In the Filters grouping, the checkboxes are announced by control type, but with no information regarding what checking/unchecking would do.  These need to read in the following manner, "Ignore special characters, checkbox, {current state}"
5. The analogous situation exists for the radio buttons in the Search Mode grouping.  You get the correct count and whether a given radio button you've landed on is checked (and since only one can be checked, that state moves as you go through them) but you get no announcement of Package name, package ID, Both, etc.
6. What looks like a slider and is being announced as such that's being used as a separator to the left of the package list has an ungodly long description being read when you land on it during traversal and it gives you almost no idea what it really is until a ton of lead-in verbiage has gone by.
7. There seems to be an invisible button overlaying the Checkbox at the top of the packages list.  If you arrive here via tab traversal the first thing you hear is "button" but activating said button does nothing.  If you tab again, you then get focus on the checkbox that "lies beneath" but it is announced only as "checkbox."  The function of said checkbox needs to be part of the announcement, and that "button that seems to do nothing" should go away.
8. The oversized "Search for packages" search edit box and associated button in the center of the empty packages list suffers from the same issue mentioned in #2 above.

I may discover more things as I have the opportunity to work more with UniGetUI.  I commend you for just how much is labeled and accessible and it's clear your intent is to have an accessible application.  Now its down to finding the small things "in hiding" in many instances that need to have labeling applied so that a screen reader announces the functions associated with these controls.

**Comment by aryanchoudharypro** [comment]
so, there are some new issues. first, the navigation items are not read by nvda like updates discover packages etc. it reads them as NavigationViewItem.  ALSO same with the packages manager and the settings buttens

**Comment by cyrmax** [comment]
@aryanchoudharypro I agree.
But as a temporary solution, if you enable tooltips reading in NVDA, then when some inaccessible element is focused, the tooltip is read sometimes.
Does work with navigation items but sometimes doesn't help with buttons.

Also the problem about package selection is still there: checkboxes to indicate whether some package is selected are not read by screen reader and are accessible only via object navigation which is considered an advanced feature.

**Comment by aryanchoudharypro** [comment]
> [@aryanchoudharypro](https://github.com/aryanchoudharypro) I agree. But as a temporary solution, if you enable tooltips reading in NVDA, then when some inaccessible element is focused, the tooltip is read sometimes. Does work with navigation items but sometimes doesn't help with buttons.
> 
> Also the problem about package selection is still there: checkboxes to indicate whether some package is selected are not read by screen reader and are accessible only via object navigation which is considered an advanced feature.


i do know that i can do that, however i keep it disabled because then it makes it extra verbose in apps like the file explorer.

**Comment by VIPPotato** [comment]
I can confirm it became less accessible than before. I didn't want to open a new issue since this one hasn't been closed.

**Comment by keyang556** [comment]
I found another accessibility issue that is still present in UniGetUI 2026.3.0.

In **Settings > Administrator privileges and other dangerous settings**, several toggle switches are announced by NVDA only as “toggle button, not pressed”, without their setting names.

This appears to affect the `SecureCheckboxCard` control. The adjacent label text is exposed separately in the UI Automation tree, but it is not associated with the `ToggleSwitch` as its accessible name. Other settings using `CheckboxCard` are announced correctly.
